test/workbox-build/node/entry-points/generate-sw-string.js

```diff
@@ -521,6 +522,57 @@ describe(`[workbox-build] entry-points/generate-sw-string.js (End to End)`, func
     });
   });
 
+  describe(`[workbox-build] behavior with 'navigationPreload'`, function() {
+    it(`should reject with a ValidationError when 'navigationPreload' is true and 'runtimeCaching' is undefined`, async function() {
+      const options = Object.assign({}, BASE_OPTIONS, {
+        navigationPreload: true,
+      });
+
+      try {
+        await generateSWString(options);
+        throw new Error('Unexpected success.');
+      } catch (error) {
+        expect(error.name).to.eql('ValidationError');
+        expect(error.details[0].context.key).to.eql('runtimeCaching');
+      }
+    });
+
+    it(`should reject with a ValidationError when 'navigationPreload' is true and 'runtimeCaching' is malformed`, async function() {
```

Comment:
This test wasn't strictly necessary but added to ensure that `joi`'s `when` syntax didn't clobber the underlying validation and figured it couldn't hurt to keep.

Reply:
👍